### PR TITLE
Feature activity history

### PR DESCRIPTION
**Pull request by kaldras**
When selecting a customer for an activity or a report, request history entries and display them at the top. A history entry fills customer > project > task with one selection.

### CODE REVIEW

File: app/components/task-selection/component.js [hunk 1]
```diff
@@ -112,6 +121,18 @@ export default Component.extend({
       return task && task.get('project.customer') || this.get('_customer')
     },
     set(value) {
+      /**
+       * It is also possible a activity/report was selected from the history.
+       * An activity/report has a duration property, check for this one.
+       */
+      if (value !== null && value.get('duration')) {
```
Comment (czosel) [inline app/components/task-selection/component.js]: Why not just `if (value && value.get('duration')) {`

File: app/components/task-selection/component.js [hunk 2]
```diff
@@ -112,6 +121,18 @@ export default Component.extend({
...
+        this.set('task', value.get('task'))
+        this.set('_customer', value.get('task.project.customer'))
+        this.set('_project', value.get('task.project'))
+
+        return this.get('_customer')
```
Comment (czosel) [inline app/components/task-selection/component.js]: Why do you have to return the customer here?

File: app/components/task-selection/component.js (hunk 1 above)
Comment (kaldras) [inline app/components/task-selection/component.js]: True that and changed

File: app/components/task-selection/component.js (hunk 2 above)
Comment (kaldras) [inline app/components/task-selection/component.js]: According to the Ember computed documentation, the computed setter should return the new value. As this is the customer property, we return the customer.

Comment (czosel) [inline app/components/task-selection/component.js]: :+1: didn't know

File: app/models/activity.js [hunk 3]
```diff
@@ -99,5 +99,23 @@ export default Model.extend({
...
+   * @type {String}
+   * @public
+   */
+  @computed('task')
+  name(task) {
+    let taskName = task.get('name')
+    let projectName = task.get('project.name')
+    let customerName = task.get('project.customer.name')
+
+    return `${customerName} > ${projectName} > ${taskName}`
```
Comment (anehx) [inline app/models/activity.js]: What happens if the task isn't set? This is theoretically not possible but I'd still handle that case somehow..

File: app/models/report.js [hunk 4]
```diff
@@ -81,5 +82,23 @@ export default Model.extend({
...
+   * @public
+   */
+  @computed('task')
+  name(task) {
+    let taskName = task.get('name')
+    let projectName = task.get('project.name')
+    let customerName = task.get('project.customer.name')
+
+    return `${customerName} > ${projectName} > ${taskName}`
+  }
```
Comment (anehx) [inline app/models/report.js]: Same as above

File: app/models/activity.js (hunk 3 above)
Comment (kaldras) [inline app/models/activity.js]: Yes, will be handled now.

File: app/models/report.js (hunk 4 above)
Comment (kaldras) [inline app/models/report.js]: Also done

**Comment by kaldras** [comment]
Was forked and finished by #30 - closing

**Pull request closed without merging**